### DISCUSSION TITLE
Workspace component - avoid syncing .bitmap file by Git

### DISCUSSION THREAD

**Discussion by davidfirst** [teambit/bit#5030] [Ideas, feature requests]
Summary of a discussion between @ranm8 , @GiladShoham  and I.

The goal is to be able to collaborate and share workspace using a new type of Bit component, which we'll call it a "workspace component". This component will contain the `.bitmap`, `workspace.jsonc` and other files that normally reside in the root repo.
One of the motivation for this feature is the difficulties we foreseen of using Git branches with Bit lanes. If .bitmap and other files don't need to be sync using Git, it'll be much easier to collaborate using Lanes.

As far as the implementation goes, this component will be pretty much like any other component, except its env will be a new core workspace-env. This way, it'll be easy to configure and distinguish this component from others. Other than that, it'll behave similar to other components, it can be tagged, exported, built and so on.

Normally, all files in the root-dir of a component are tracked by Bit. This component has the root dir of the workspace itself, so in order to not track all the components files to this component, we'll implement a new feature that exclude all root-dirs of other components when tracking a component. 

To exclude other files that are not components, we'll need to implement another feature, which let users configure in the variants, file exclusion rules. To refer to this workspace-component in the `variants` section, we'll use `.` notation, to indicate the root-dir. 
To improve it even more, we'll let users create `.bitignore` file. In the root level for the workspace component and in the component's directories. 

When tagging components, this workspace-component needs to be tagged as well because the .bitmap file had changed. However, normally, the .bitmap file gets changed by the end of the process, so if we tag it as other component, the .bitmap won't be up to date. To mitigate this, we'll tag this specific component after other components were successfully tagged.

**Comment by GiladShoham** [comment]
Few more notes:
We will move the bitmap file into the bit folder so it won't be tracked by git anymore.

We will have new commands (names tbd)
Bit clone - will get a ws component id and will import and checkout the ws component and all the components needed.
This will be run on a new empty folder.

Bit pull will fetch and checkout the ws component then will checkout all the other components.
It's basically a sugar of import the ws component check it out. Then checkout the other components (as the bitmap changed as a result of check out the workspace component)

We will add a prop in the ws.jsonc to set the ws component id (to support fresh git clone followed by bit pull. In that case you don't the bitmap yet so you need the id)

**Comment by davidfirst** [comment]
Another thought we were having is move the `.bitmap` file from the workspace to the scope, inside `.bit`. this way, in "bit flow" (vs. "git flow"), it won't be visible to the users.
Problem is, what happens if a user deletes the `.bit` directory. It'll loose the location of the components on the filesystem.
